### ISSUE TITLE
Fix RestAPI Notification block off by 1 error

### ISSUE THREAD

**Issue by ixje** [CityOfZion/neo-python#554]
### Current behavior
See https://github.com/CityOfZion/neo-python/issues/478#issuecomment-412373486

the notification DB was created while the `block-1` bug was present, so while a fix is already merged into `neo-python` the test never failed. Correct behaviour is notification is in the same block number as the tx.

### Expected behavior
Correct at all indexing i.e.
https://github.com/CityOfZion/neo-python/blob/ba815b870614d01f9295d934c8dc3a172765e1aa/neo/api/REST/RestApi.py#L159

### How to reproduce

see issue mentioned above

### Your environment

Let us know in what environment you're running into the issue:

- OS: (Windows version/Linux distro/OSX version) all
- neo-python version: (hint: use `np-prompt --version`) 0.7.7-dev
- Python version: (hint: `python -V`) 3.6.5

**Comment by jseagrave21** [comment]
@ixje I did some troubleshooting with the error I noticed in JSON RPC. Removing the "-1" in RestApi.py does not affect the errant test in test_json_rpc_api.py where I query block 10 and `getblock` returns block 9. I dug down to neo.core.blockchain but wasn't able to figure out why the error is happening. Should I create a separate issue? I do feel like these problems are related. In the meantime I will continue troubleshooting.
